hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationBackupHandler.java

```diff
@@ -333,13 +341,25 @@ private static Backup newBackup(BackupAwareOperation backupAwareOp, Object backu
         return backup;
     }
 
-    /**
-     * Verifies that the backup of a partition doesn't end up at the member that also has the primary.
-     */
-    private void assertNoBackupOnPrimaryMember(InternalPartition partition, Address target) {
-        if (target.equals(node.getThisAddress())) {
-            throw new IllegalStateException("Normally shouldn't happen! Owner node and backup node "
-                    + "are the same! " + partition);
+    private boolean skipSendingBackupToTarget(InternalPartition partition, PartitionReplica target) {
+        ClusterServiceImpl clusterService = node.getClusterService();
+
+        assert !target.isIdentical(nodeEngine.getLocalMember()) : "Could not send backup operation, because "
```

Comment:
How come we don't throw the exception anymore?

Reply:
Because that's not a possible case unless we create that scenario artificially, that's why I converted it to an assertion.